src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ERR.cs

```diff
@@ -14,7 +14,7 @@ internal static partial class Crypto
         internal static extern ulong ErrClearError();
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_ErrGetErrorAlloc")]
-        private static extern ulong ErrGetErrorAlloc([MarshalAs(UnmanagedType.Bool)] out bool isAllocFailure);
+        private static extern unsafe ulong ErrGetErrorAlloc(int* isAllocFailure);
```

Comment:
> Should we revisit the preivously stated position on [Out] and use it for communicating such things?

I'm dubious of doing this. The interop boundary should and is always not going to be look as good as normal C# and I am of the perspective that is fine. Adding the `[Out]` is helpful but no more so than a comment `/* out */` &ndash; which I am not against. Also, the `out` doesn't really guarantee anything at the interop boundary because there is nothing enforcing the native side from doing it so it sort of leads people into a false sense of security.

Reply:
My top-level point is that these changes are making the code less maintainable.  If this were a GeneratedDllImport, presumably for a signature like:
```C#
private static partial ulong ErrGetErrorAlloc([MarshalAs(UnmanagedType.Bool)] out bool isAllocFailure);
```
the generator will spit out code like:
```C#
private static partial unsafe ulong ErrGetErrorAlloc(out bool isAllocFailure)
{
    int isAllocFailure_Local;
    ulong result = _ErrGetErrorAlloc(&isAllocFailure_Local);
    isAllocFailure = isAllocFailure_Local != 0;
    return result;
}
```
which should be inlineable and avoids all callers having to effectively write the same thing repeatedly while also making the interop harder to misuse.  One dev decides at the time of authoring the P/Invoke how things map, and then all devs consuming it from then on in the project don't need to be as concerned about it.

Am I off-base?

Reply:
@stephentoub I don't think you are off base in general &ndash; make `GeneratedDllImport` the way to do things. I simply don't think we should force that in all cases. In the runtime as the high order bit always seems to be performance and trimming as much off as possible always seems to be the goal. Additionally, I don't think adding another function is always appropriate because it does rely on JIT details that are difficult to track down and I am not sure Mono has the same optimizations that RyuJIT does. We know that a fully blittable API will be inlined so let's use that when we can and avoid the code gen when we don't need it. There is nothing stopping anyone from doing exactly as suggested above but in the runtime our goal was to remove absolutely any extra ceremony when possible.

Reply:
Move offline - 21.